pkg/apis/serving/v1alpha1/trainedmodel_webhook_test.go

```diff
@@ -0,0 +1,93 @@
+package v1alpha1
+
+import (
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func makeTestTrainModel() TrainedModel {
+	tm := TrainedModel{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "TrainedModel",
+			APIVersion: "v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "bar",
+		},
+		Spec: TrainedModelSpec{
+			InferenceService: "Parent",
+			Model: ModelSpec{
+				StorageURI: "example.com/sklearn/iris",
+				Framework:  "sklearn",
+			},
+		},
+	}
+
+	quantity, err := resource.ParseQuantity("100Mi")
+	if err == nil {
+		tm.Spec.Model.Memory = quantity
+	}
+
+	return tm
+}
+
+func TestTrainedModelCreation(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	tm := makeTestTrainModel()
+	g.Expect(tm.ValidateCreate()).Should(gomega.Succeed())
+}
+
+func TestUpdateMutableSpec(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	tm := makeTestTrainModel()
+	old := tm.DeepCopyObject()
+	tm.Spec.InferenceService = "parent2"
+	tm.Spec.Model.StorageURI = "example2.com/sklearn/iris"
+	tm.Spec.Model.Framework = "sklearn2"
+	g.Expect(tm.ValidateUpdate(old)).Should(gomega.Succeed())
+}
+
+func TestUpdateWithInvalidName(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	tm := makeTestTrainModel()
+	old := tm.DeepCopyObject()
+	tm.Name = "1abc"
+	g.Expect(tm.ValidateUpdate(old)).ShouldNot(gomega.Succeed())
+}
+
+func TestUpdateImmutableMemory(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	tm := makeTestTrainModel()
+	old := tm.DeepCopyObject()
+	tm.Spec.Model.Memory = resource.Quantity{Format: "300Mi"}
+	g.Expect(tm.ValidateUpdate(old)).ShouldNot(gomega.Succeed())
+}
+
```

Comment:
Can we switch to use `table testing` which is golang's preferred testing method ?
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests